src/components/AddAssignmentModal.js

```diff
@@ -0,0 +1,132 @@
+import React from "react";
+import Modal from "@material-ui/core/Modal";
+import { makeStyles } from "@material-ui/core/styles";
+import TextField from "@material-ui/core/TextField";
+import Button from "@material-ui/core/Button";
+import Cookies from 'js-cookie';
+import {SERVER_URL} from '../constants.js';
+import { ToastContainer, toast } from 'react-toastify';
+
+const useStyles = makeStyles((theme) => ({
+  paper: {
+    position: "absolute",
+    width: 400,
+    backgroundColor: theme.palette.background.paper,
+    boxShadow: theme.shadows[5],
+    padding: theme.spacing(2, 4, 3),
+    top: "50%",
+    left: "50%",
+    transform: "translate(-50%, -50%)",
+  },
+  textField: {
+    marginBottom: theme.spacing(2),
+  },
+  button: {
+    margin: theme.spacing(1),
+  },
+}));
+
+const AddAssignmentModal = () => {
+  const classes = useStyles();
+  const [open, setOpen] = React.useState(false);
+  const [assignmentName, setAssignmentName] = React.useState("");
+  const [courseID, setCourseID] = React.useState("");
+  const [dueDate, setDueDate] = React.useState("");
+
+  const handleOpen = () => {
+    setOpen(true);
+  };
+
+  const handleClose = () => {
+    setOpen(false);
+  };
+
+  const handleSubmit = () => {    
+    console.log("you added: " + courseID);
+
+    const token = Cookies.get('XSRF-TOKEN');
+      fetch(`${SERVER_URL}/assignments/new?name=${assignmentName}&due=${dueDate}&id=${courseID}`, 
```

Comment:
i havent read the assignment yet, but cookies are sent with every request automatically, so generally in the real world you don't need to additionally send a header with the xsrf token manually attatched.

may be needed for the assignment though